fastlane/lanes/build.rb

```diff
@@ -134,6 +134,32 @@
     end
   end
 
+  #####################################################################################
+  # download_signed_apk_from_google_play
+  # -----------------------------------------------------------------------------------
+  # This lane downloads the signed apk from Play Store for the given version and app
+  # -----------------------------------------------------------------------------------
+  # Usage:
+  # bundle exec fastlane download_signed_apk_from_google_play app:<wordpress|jetpack> version:<versionName,versionCode>
+  #####################################################################################
+  lane :download_signed_apk_from_google_play do |options|
+    app = get_app_name_option!(options)
+    package_name = APP_SPECIFIC_VALUES[app.to_sym][:package_name]
+    version = options[:version] || android_get_release_version() # default to current release version
+
+    if version.is_a?(String) # for when calling from command line
+      (version_name, version_code) = version.split(',')
+      version = { 'name' => version_name, 'code' => version_code || '1' }
```

Comment:
In practice, for this specific use case, we probably don't want to fallback to using a `versionCode` of `1` because the GooglePlay API will really require the real versionCode to know what APK to download.
```suggestion
      UI.user_error!('Please pass the `version` option as a comma-separated `name,code` value') if version_code.nil?
      version = { 'name' => version_name, 'code' => version_code }
```

Reply:
@AliSoftware I have been downloading the APK by running `bundle exec fastlane download_signed_apk_from_google_play app:jetpack` which uses the version name & version code for the current branch. This has been working very well for me.

Could you elaborate on your concerns about having a default value? I feel like it's a valid case, but maybe I am missing something?

Reply:
Oh, this is in a completely different place. Sorry @AliSoftware - I shouldn't work before I put my lenses on. I'll take a look at this later today. 🤦

Reply:
My note was about if you provide a `version:` but without the expected comma-separate values format, like `bundle exec fastlane download_signed_apk_from_google_play app:jetpack version:22.9-rc-1`.

In such cases, the branch with `is_a?(String)` would be taken for the version value privided by command line and `split` call would extract `22.9-rc-1` into `version_name`, but leave `version_code` as `nil` because you didn't provide a comma + version code in command line option. In such case, that default value of `1` for `version_code` would make it behave like if you provided `version:22.9-rc-1,1` to the command line (instead of `22.9-rc-1,1359`)

---

To avoid that case, my code suggestion above would just error if you accidentally tried to provide a `version:` parameter from the command line but without a comma.

Alternatively you could consider reading the `versionCode` from the current version file (while having the versionName be overriden by the value passed as `version:22.9-rc-1` parameter)… but this doesn't seem worth it to cover those twisted cases — especially when I don't expect you to provide the `version:` option via the command line in 99% of use cases of this lane anyway. And also, all those edge cases when providing the value from command line would go away if you decide to refactor the lane options in a subsequent PR at some point to pass those as two separate parameters to make the whole thing less convoluted, [as discussed here](https://github.com/wordpress-mobile/WordPress-Android/pull/18813#discussion_r1277360443)… so probably not worth spending more time on it than just the basic `UI.user_error!` protection I suggested above to avoid accidental misuse?

Reply:
@AliSoftware You probably missed [my last comment](https://github.com/wordpress-mobile/WordPress-Android/pull/18813#discussion_r1277764744), but I deserve that wall of text. Sorry about wasting your time 😢

Reply:
No worries 🙂  🤗  ☕  👓

Reply:
Now that I can actually see what the suggestion is, this makes a lot of sense. Thank you for the suggestion and your understanding about the earlier mistake @AliSoftware 🙇‍♂️ 

P.S: This has not been a good morning so far, just after posting that comment, I also accidentally merged `trunk` into this branch instead of the `release/22.9` which is why there is a force push about an hour ago. Hopefully rest of the day will be better 🤞

Reply:
![virtual-hug](https://github.com/wordpress-mobile/WordPress-Android/assets/216089/3f35aad3-56d8-45b7-8747-7d90115d7059)